tests/Doctrine/Tests/DBAL/Functional/Schema/SchemaManagerFunctionalTestCase.php

```diff
@@ -1165,4 +1173,43 @@ public function testDoesNotListIndexesImplicitlyCreatedByForeignKeys()
         $this->assertArrayHasKey('explicit_fk1_idx', $indexes);
         $this->assertArrayHasKey('idx_3d6c147fdc58d6c', $indexes);
     }
+
+    public function testCreateSchemaOnLargeNumberOfTables(): void
+    {
+        $numberOfTablesCreated = $this->createLargeNumberOfTables();
+
+        // Introspect the db schema.
+        $startTime = microtime(TRUE);
+        $schema = $this->_sm->createSchema();
+        $endTime = microtime(TRUE);
+
+        $this->assertGreaterThanOrEqual($numberOfTablesCreated, count($schema->getTables()));
+        $this->assertLessThan($this->timeThresholdForCreateSchema, $endTime - $startTime, 'createSchema() executed in less than ' . $this->timeThresholdForCreateSchema . ' sec.');
```

Comment:
This test is by design brittle since the time spent in userland on the execution depends on a lot of factors we're not in control of. Be ready to have this test randomly failing.